### ISSUE TITLE
Known performance issue? 

### ISSUE THREAD

**Issue by rafaelugolini** [saltyshiomix/nextron#15]
Looking at the README.md you wrote:

`If you need more performance with Electron, you should see these boilerplates`

Is there any known performance issue?

**Comment by saltyshiomix** [comment]
Hi @rafaelugolini , thank you for your question :)

There is no performance issues for now, but best performance is SPA in renderer process so I mention it.
(Next.js is a SSR framework and is not supported SPA. I'm looking for SPA techniques with Next.js, too.)

**Comment by rafaelugolini** [comment]
@saltyshiomix I just tested using `<Link href="..." prefetch>...</Link>` and you can feel the links happening faster. 
https://nextjs.org/docs/#prefetching-pages

Also worth investigating static export for production, right?
https://nextjs.org/docs/#static-html-export

Do you want help with this? Maybe I can create a PR about it.

**Comment by saltyshiomix** [comment]
@rafaelugolini 

PRs are so much welcome 🥇 

But please make sure that the electron in production mode can handle **only** static files so the url changes comparing to development mode.
I'm not sure prefetching-pages works when production mode :(

Nextron resolve these problems by ipc communication when facing difficulties.

For more information:

**main process: resolveWithIpc()**
https://github.com/saltyshiomix/nextron/blob/master/examples/_template/ts/main/helpers/resolveWithIpc.ts

**renderer process: resolveWithIpc()**
https://github.com/saltyshiomix/nextron/blob/master/examples/_template/ts/renderer/helpers/resolveWithIpc.ts

Digging deeper: ipc communication example:

```bash
$ nextron init test-app --template ipc-communication
```

**Comment by rafaelugolini** [comment]
@saltyshiomix I was looking at some projects (e.g., https://github.com/zeit/now-desktop), and they use routes to separate windows (about, feed, ...)

Maybe it's a matter on how to organize the app and not depend on next.js routes?

Do you have any ideas how to solve this?

**Comment by saltyshiomix** [comment]
@rafaelugolini 

Good question.
As you pointed out, nextron's structure is not equal to now-desktop one.

The now-desktop defines routes in main process like this:

https://github.com/zeit/now-desktop/blob/canary/main/index.js#L178

```js
const windows = {
  main: mainWindow(tray),
  tutorial: tutorialWindow(tray),
  about: aboutWindow(tray)
}

// Make the window instances accessible from everywhere
global.windows = windows
```

And accesses routes from renderer process like this:

https://github.com/zeit/now-desktop/blob/canary/renderer/pages/about.js#L127

```js
const windows = this.remote.getGlobal('windows')

windows.tutorial.show()
```

Because electron in production mode (binary) can handle only static files for security reasons, we need get routes by accessing main process. (ex: ipc communication, electron.remote and so on)

Here is an example code that shows a difference between dev and prod mode in electron:

https://github.com/zeit/now-desktop/blob/canary/main/index.js#L178

```js
const loadPage = (win, page) => {
  if (isDev) {
    win.loadURL(`http://localhost:8000/${page}`)
  } else {
    win.loadFile(`${electron.app.getAppPath()}/renderer/out/${page}/index.html`)
  }
}
```

Nextron's one:

https://github.com/saltyshiomix/nextron/blob/master/examples/_template/js/main/background.js#L20

```js
if (isProd) {
  const homeFile = join(app.getAppPath(), 'app/home/index.html')
  mainWindow.loadFile(homeFile)
} else {
  const homeUrl = 'http://localhost:8888/home'
  mainWindow.loadURL(homeUrl)
  mainWindow.webContents.openDevTools()
}
```

In conclusion, though I want to implement pretty routing, we need to handle routes both dev and prod mode.

So I think `<Link prefetch>` works only development mode.

I want help, too. I'm sorry not helps.

**Comment by saltyshiomix** [comment]
@rafaelugolini 

I found it:
https://github.com/GoogleChromeLabs/quicklink

It says:

> Todo: Explore detecting file-extension of resources and using rel=preload for high priority fetches

It may be some helps.

**Comment by saltyshiomix** [comment]
There are no critical performance issues for one year, so I think it's time for closing this issue.
